xc/xc7/tests/soc/litex/mini/CMakeLists.txt

```diff
@@ -1,18 +1,38 @@
-add_file_target(FILE mem.init)
-add_file_target(FILE mem_1.init)
-add_file_target(FILE mem_2.init)
+add_file_target(FILE arty_clocks.xdc)
 
-add_fpga_target(
+set(SOURCES
+  top.v
+  top.pcf
+  top.sdc
+  mem.init
+  mem_1.init
+  mem_2.init
+)
+
+add_litex_test(
   NAME minilitex_arty
+  LITEX_DIR arty_soc
+  LITEX_BOARD a7-35
+  LITEX_SOURCES ${SOURCES}
+  EXTERNAL_SOURCES ${VEXRISCV}
   BOARD arty-full
-  SOURCES
-    minilitex_arty.v
-    VexRiscv_Lite.v
-  INPUT_IO_FILE arty.pcf
-  INPUT_XDC_FILE minilitex.xdc
+  GENERATE_SCRIPT ${GENERATE_LITEX}
+  FIXUP_SCRIPT ${FIXUP_XDC}
+  FLAGS
+    --toolchain symbiflow
+  VIVADO_XDC arty_clocks.xdc
```

Comment:
Why are we not using the XDC from LiteX?  Or does LiteX no supply a clock constraint?

Reply:
LiteX does supply clock constraints, but the problem was, IIRC, that the output of fasm2bels (Verilog/XDC) does not include certain clock ports present in the LiteX generated XDC, ending up in an error when trying to read it.

I need to double-check this though.

Reply:
This is still outstanding?

Reply:
I will open an issue to keep track of this

Reply:
In general we need a burn down for the remaining "weirdness" around the LiteX support.  Examples include this, the `fixup_xdc` script, the `--toolchain symbiflow`/SDC stuff from the other converstation, etc.

Reply:
Oh and the bittool / diff FASM disable too.